backends/s3/s3testing/minio.go

```diff
@@ -0,0 +1,70 @@
+package s3testing
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"time"
+)
+
+const (
+	AdminUserOrPassword = "simpleblob"
+)
+
+var ErrMinioNotFound = exec.ErrNotFound
+
+// ServeMinio starts a Minio server in the background,
+// and waits for it to be ready.
+// It returns its address,
+// and a function to stop the server gracefully.
+//
+// The admin username and password for the server are both "simpleblob".
+//
+// If the minio binary cannot be found in PATH,
+// ErrMinioNotFound will be returned.
+func ServeMinio(ctx context.Context, dir string) (string, func() error, error) {
+	cmdname, err := exec.LookPath("minio")
+	if err != nil {
+		return "", nil, err
+	}
+
+	port, err := FreePort()
+	if err != nil {
+		return "", nil, err
+	}
+	addr := fmt.Sprintf("127.0.0.1:%d", port)
+
+	cmd := exec.CommandContext(ctx, cmdname, "server", "--quiet", "--address", addr, dir)
+	cmd.Env = append(os.Environ(), "MINIO_BROWSER=off")
```

Comment:
Nit: I had to check the `os.Environ()` implementation to check if this is safe and would not update a global slice. It happens to create a new `[]string` for every call, but it would better to no relay on this:

```go
cmd.Env = append([]string{}, os.Environ()...)
cmd.Env = append(cmd.Env, "MINIO_BROWSER=off")
```